lib/client/auth-code.js

```diff
@@ -18,7 +19,13 @@ module.exports = function (config) {
     params.response_type = 'code';
     params.client_id = config.clientID;
 
-    return config.site + config.authorizationPath + '?' + qs.stringify(params);
+    var isAbsolute = String(Url.parse(config.authorizationPath).protocol)
```

Comment:
'url.resolve' will resolve a url like a browser would, but if the 'site' were to be configured as 'http://domain.com/oauth' and the 'authorizationPath' were '/authorize' it will drop the '/oauth' from the path.

Apparantly the test suite does not account for this edge-case, hence I didn't see it failing before.